prow/sidecar/censor.go

```diff
@@ -418,10 +419,79 @@ func loadSecrets(paths []string) ([][]byte, error) {
 				return err
 			}
 			secrets = append(secrets, raw)
+			var parser = func(bytes []byte) ([]string, error) {
+				return []string{}, nil
+			}
+			if info.Name() == ".dockercfg" {
+				parser = loadDockercfgAuths
+			}
+			if info.Name() == ".dockerconfigjson" {
+				parser = loadDockerconfigJsonAuths
+			}
+			extra, parseErr := parser(raw)
+			if parseErr != nil {
+				return fmt.Errorf("could not read %s as a docker secret: %v", path, parseErr)
+			}
+			// It is important that these are added to the list of secrets *after* their parent data
+			// as we will censor in order and this will give a reasonable guarantee that the parent
+			// data (a superset of any of these fields) will be censored in its entirety, first. It
+			// remains possible that the sliding window used to censor pulls in only part of the
+			// superset and some small part of it is censored first, making the larger superset no
+			// longer match the file being censored.
+			for _, item := range extra {
+				secrets = append(secrets, []byte(item))
+			}
 			return nil
 		}); err != nil {
 			return nil, err
 		}
 	}
 	return secrets, nil
 }
+
+// loadDockercfgAuths parses auth values from a kubernetes.io/dockercfg secret
+func loadDockercfgAuths(content []byte) ([]string, error) {
```

Comment:
Could you explain why we need this? In case someone parses and extracts auth items from a pull secret?

Reply:
Explain here in a comment or in the code? Yes, it's to guard the items if they're extracted. Or imagine a tool that shows part of the output of a test, which cuts a secret in half - this helps us censor the important bits even if we can't do the whole thing.